### PR TITLE
Add a note for the surprising nature of switch scopes

### LINKED ISSUE #945: Add note for switch scope oddity
This was a comment in #942. The scope of variables introduced in switch statements is odd, and it would be good to draw the reader's attention to it.

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #945.

### CODE REVIEW

File: standard/basic-concepts.md [hunk 1]
```diff
@@ -92,6 +92,25 @@ There are several different types of declaration spaces, as described in the fol
   - The syntactic translation of a *query_expression* ([§12.20.3](expressions.md#12203-query-expression-translation)) may introduce one or more lambda expressions. As anonymous functions, each of these creates a local variable declaration space as described above.
 - Each *block* or *switch_block* creates a separate declaration space for labels. Names are introduced into this declaration space through *labeled_statement*s, and the names are referenced through *goto_statement*s. The ***label declaration space*** of a block includes any nested blocks. Thus, within a nested block it is not possible to declare a label with the same name as a label in an enclosing block.
 
+> *Note*: The fact that variables declared directly within a *switch_section* are added to the local variable declaration space of the *switch_block* instead of the *switch_section* can lead to surprising code. In the example below, the local variable `y` is in scope within the switch section for case 1, despite the declaration appearing in the switch section for case 0.
```
Comment (KalleOlaviNiemitalo) [inline standard/basic-concepts.md]: "the switch section for case 1" feels a little odd when it is is `default:` rather than `case 1:`.

Comment (jskeet) [reply standard/basic-concepts.md]: Whoops, thanks. Changed the code without changing the note. Fixing.

**Review by BillWagner — APPROVED** [review]
This also LGTM @jskeet 

Let's :shipit:

**Comment by jskeet** [comment]
Let's wait until the meeting to just get more eyes before shipping (on this and others).

**Review by gafter — APPROVED** [review]

**Review by Nigel-Ecma — APPROVED** [review]

**Pull request merged**